lib/winston/transports/console.js

```diff
@@ -18,6 +18,13 @@ const TransportStream = require('winston-transport');
  * @extends {TransportStream}
  */
 module.exports = class Console extends TransportStream {
+  // Keep a reference to the log, warn, and error console methods
+  // in case they get redirected to this transport after the logger is
+  // instantiated. This prevents a circular reference issue.
+  _consoleLog = console.log;
```

Comment:
Minor potential nitpick, which I'm 99% certain doesn't matter on any Node.js version, but it should probably be `_consoleLog = console.log.bind(console);`, since otherwise it's the equivalent of losing the `this` in the invocation / calling `console.log.call(undefined, ...)`, which throws "Invalid invocation" on old browsers (but, as far as I know, not actually on any Node.js version, and Winston is never ran in browsers, or, at least, not ancient ones?).

Reply:
Done!